pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -125,9 +127,10 @@ var _ reconcile.Reconciler = &ReconcileKogitoApp{}
 type ReconcileKogitoApp struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	client *kogitocli.Client
-	scheme *runtime.Scheme
-	cache  cachev1.Cache
+	client             *kogitocli.Client
+	scheme             *runtime.Scheme
+	cache              cachev1.Cache
+	isImagestreamReady bool
```

Comment:
It seems this variable is only used in function `ensureKogitoImageStream`, it could be just a local variable.

Reply:
Sorry, I didn't notice the comments `set the isImagestreamReady to true to avoid it to be checked every time the reconcile runs`.

But I don't think we want to maintain any state by the `Reconciler`. [It should re-evaluate the entire state as opposed to just what changed](https://blog.openshift.com/kubernetes-operators-best-practices/). Otherwise, we will have to figure out how to sync the states between the `Reconciler` and the cluster. 

If you think the checking is a waste in most cases, maybe we can just check the single image stream used by this particular kogito app and create only that single image stream. Because in many cases, only one image stream will used by one kogito app. It is not very helpful to create all the image streams.

Reply:
I liked the idea of creating just the image**s** that would be used by this deployment.
And also, wait for the object to be create is a best practice indeed, thanks for pointing this out @xiezhang7 

Instead, create the ImageStream and return a Reconcile Request after 5/10 seconds and them your algorithm will check for the existence of the IS. If it's not there, reconcile again.

Always with a message to the user: "ImageStream not ready/found, scheduling reconcile".

Thanks!